test/e2e/network/service.go

```diff
@@ -2730,6 +2730,98 @@ var _ = common.SIGDescribe("Services", func() {
 		framework.ExpectNoError(err, "failed to delete Service %v in namespace %v", testService.ObjectMeta.Name, ns)
 		framework.Logf("Service %s deleted", testSvcName)
 	})
+
+	ginkgo.It("should delete a collection of services", func() {
+
+		ns := f.Namespace.Name
+		svcClient := f.ClientSet.CoreV1().Services(ns)
+		svcResource := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "services"}
+		svcDynamicClient := f.DynamicClient.Resource(svcResource).Namespace(ns)
+
+		svcName := "e2e-svc-" + utilrand.String(5)
+		svcLabels := map[string]string{"e2e-test-service": "true"}
+		labelSelector := labels.SelectorFromSet(svcLabels).String()
+
+		w := &cache.ListWatch{
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				options.LabelSelector = labelSelector
+				return svcClient.Watch(context.TODO(), options)
+			},
+		}
+
+		svcList, err := cs.CoreV1().Services("").List(context.TODO(), metav1.ListOptions{LabelSelector: labelSelector})
```

Comment:
list all services or only the ones in the test namespace?

Reply:
That `svcList` (filtered by the label selctor) is getting setup so I can access the resource version later in line 2777 

https://github.com/kubernetes/kubernetes/blob/8158c1a316cf8c8742a6972b70bd2d770d62f148/test/e2e/network/service.go#L2777

Reply:
I was wondering why do you prefer to list in all namespaces instead of in the test namespace
```
svcList, err := cs.CoreV1().Services(ns).List
```

Reply:
Just the part of the initial watch pattern I've been using. I'll give the above option a go.

Reply:
@aojea Updated `svcList` to use the test namespace now.